http-scala-fx/src/main/scala/fx/Http.scala

```diff
@@ -0,0 +1,329 @@
+    DELETE[A](Option(timeout), 0, headers: _*)
+
+  @tailrec
+  private def DELETE[A](
+      maybeTimeout: Option[ScalaDuration],
+      retryCount: Int = 0,
+      headers: HttpHeader*): (HttpResponseMapper[A]) ?=> Http[HttpResponse[A]] =
+    val mapper = summon[HttpResponseMapper[A]]
+    val config = summon[HttpClientConfig]
+    val retries = config.maximumRetries.getOrElse(summon[HttpRetries])
+    val requestBuilder =
+      addHeadersToRequestBuilder(HttpRequest.newBuilder().DELETE().uri(uri), headers: _*)
+    val response =
+      handle(
+        summon[HttpClient]
+          .client
+          .send(
+            maybeTimeout
+              .fold(requestBuilder) { fd =>
+                requestBuilder.timeout(Duration.ofMillis(fd.toMillis))
+              }
+              .build(),
+            mapper.bodyHandler)) { ex => HttpExecutionException(ex).shift }
+    if (response.shouldRetry(retryCount)) {
+      DELETE(maybeTimeout, retryCount + 1, headers: _*)
+    } else {
+      response
+    }
+
+  def OPTIONS[A](headers: HttpHeader*): (HttpResponseMapper[A]) ?=> Http[HttpResponse[A]] =
+    OPTIONS(None, 0, headers: _*)
+
+  def OPTIONS[A](
+      timeout: ScalaDuration,
+      headers: HttpHeader*): (HttpResponseMapper[A]) ?=> Http[HttpResponse[A]] =
+    OPTIONS(Option(timeout), 0, headers: _*)
+
+  @tailrec
+  private def OPTIONS[A](
+      maybeTimeout: Option[ScalaDuration],
+      retryCount: Int = 0,
+      headers: HttpHeader*): (HttpResponseMapper[A]) ?=> Http[HttpResponse[A]] =
+    val mapper = summon[HttpResponseMapper[A]]
+    val config = summon[HttpClientConfig]
+    val retries = config.maximumRetries.getOrElse(summon[HttpRetries])
+    val requestBuilder = addHeadersToRequestBuilder(
+      HttpRequest.newBuilder().method("OPTIONS", BodyPublishers.noBody).uri(uri))
+    val response =
+      handle(
+        summon[HttpClient]
+          .client
+          .send(
+            maybeTimeout
+              .fold(requestBuilder) { fd =>
+                requestBuilder.timeout(Duration.ofMillis(fd.toMillis))
+              }
+              .build(),
+            mapper.bodyHandler)) { ex => HttpExecutionException(ex).shift }
+    if (response.shouldRetry(retryCount)) {
+      OPTIONS(maybeTimeout, retryCount + 1, headers: _*)
+    } else response
+
+  def TRACE[A](headers: HttpHeader*): HttpResponseMapper[A] ?=> Http[HttpResponse[A]] =
+    TRACE[A](None, 0, headers: _*)
+
+  def TRACE[A](
+      timeout: ScalaDuration,
+      headers: HttpHeader*): HttpResponseMapper[A] ?=> Http[HttpResponse[A]] =
+    TRACE[A](Option(timeout), 0, headers: _*)
+
+  @tailrec
+  private def TRACE[A](
+      maybeTimeout: Option[ScalaDuration],
+      retryCount: Int = 0,
+      headers: HttpHeader*): HttpResponseMapper[A] ?=> Http[HttpResponse[A]] =
+    val mapper = summon[HttpResponseMapper[A]]
+    val config = summon[HttpClientConfig]
+    val retries = config.maximumRetries.getOrElse(summon[HttpRetries])
```

Comment:
```suggestion
```

Reply:
Same as above.

Reply:
Resolved in [ce2b2a4](https://github.com/47deg/TBD/pull/37/commits/ce2b2a4da2a01b983a46e3a5ab23f80770d74d24)